trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/CHandler.java

```diff
@@ -2714,6 +2718,19 @@ public static CStorageClass scConstant2StorageClass(final int storageClass) {
 		}
 	}
 
+	private ExpressionResult makeAtomicAssignmentIfNecessary(final ILocation loc, final LRValue leftHandSide,
+			final List<Statement> nonAtomicStatements, final ExpressionResult rhs, final IASTNode hook) {
+		final ExpressionResult assignment = makeAssignment(loc, leftHandSide, Set.of(), rhs, hook);
+		if (!leftHandSide.getCType().isAtomic()) {
+			return assignment;
+		}
+		final List<Statement> atomicStatements = new ArrayList<>(assignment.getStatements());
+		atomicStatements.removeAll(nonAtomicStatements);

```

Comment:
As I understand it, the expectation is that `nonAtomicStatements` is a prefix of the list `atomicStatements`. In that case, please add a check here that this indeed holds, and remove the first `nonAtomicStatements.size()` statements.

Reply:
Good idea, I will add this!

Reply:
But what do you mean with `nonAtomicStatements.size()`?